config/configgrpc/configgrpc.go

```diff
@@ -218,6 +219,21 @@ func NewDefaultServerConfig() ServerConfig {
 }
 
 func (cc *ClientConfig) Validate() error {
+	grpcOpts, err := cc.getGrpcDialOptions(context.Background(), componenttest.NewNopHost(), componenttest.NewNopTelemetrySettings())
+	if err != nil {
+		return err
+	}
+	// Perform a simple check that we will be able to create a new client
+	conn, err := grpc.NewClient(cc.Endpoint, grpcOpts...)
```

Comment:
The documentation for `NewClient` explicitly states that no I/O is performed, so I opted to use that here instead of `ToClientConn`, since I don't know if the same can be said for `DialContext`.